### ISSUE TITLE
Guidance on how to start JupyterHub with SSL

### ISSUE THREAD

**Issue by cenveoanalyst** [jupyter/jupyterhub#451]
Greetings!

I have jupyterhub working fine on localhost with PAM authentication and am quite excited about moving forward with deployment in my organization. The documentation strongly suggests running jupyterhub with SSL. When I try running the ip/ssl example from the documentation:

```
dude1@dude1-VirtualBox:~/Documents$ ls
ca.cert     certtest.perm         jupyterhub_config.py      jupyterhub.sqlite
ca.key4096  First Notebook.ipynb  jupyterhub_cookie_secret  keytest.key
dude1@dude1-VirtualBox:~/Documents$ sudo jupyterhub --ip 10.0.1.2 --port 443 --ssl-key ca.key4096 --ssl-cert ca.cert
[sudo] password for dude1: 
[I 2016-03-03 16:25:03.624 JupyterHub app:523] Loading cookie_secret from /home/dude1/Documents/jupyterhub_cookie_secret
[W 2016-03-03 16:25:03.645 JupyterHub app:258] 
    Generating CONFIGPROXY_AUTH_TOKEN. Restarting the Hub will require restarting the proxy.
    Set CONFIGPROXY_AUTH_TOKEN env or JupyterHub.proxy_auth_token config to avoid this message.
    
[W 2016-03-03 16:25:03.659 JupyterHub app:642] No admin users, admin interface will be unavailable.
[W 2016-03-03 16:25:03.659 JupyterHub app:643] Add any administrative users to `c.Authenticator.admin_users` in config.
[I 2016-03-03 16:25:03.659 JupyterHub app:669] Not using whitelist. Any authenticated user will be allowed.
[I 2016-03-03 16:25:03.673 JupyterHub app:1052] Hub API listening on http://localhost:8081/hub/
[C 2016-03-03 16:27:10.921 JupyterHub app:1058] Failed to start proxy
    Traceback (most recent call last):
      File "/usr/local/lib/python3.4/dist-packages/jupyterhub/app.py", line 1056, in start
        yield self.start_proxy()
      File "/usr/local/lib/python3.4/dist-packages/jupyterhub/app.py", line 770, in start_proxy
        if self.proxy.public_server.is_up() or self.proxy.api_server.is_up():
      File "/usr/local/lib/python3.4/dist-packages/jupyterhub/orm.py", line 117, in is_up
        socket.create_connection((self.ip or localhost(), self.port))
      File "/usr/lib/python3.4/socket.py", line 512, in create_connection
        raise err
      File "/usr/lib/python3.4/socket.py", line 503, in create_connection
        sock.connect(sa)
    TimeoutError: [Errno 110] Connection timed out
```

I get the following error. I notice the HUB API is not listening on SSL. Does the HUB API need to be? I'm obviously missing something here so any tips?

**Comment by minrk** [comment]
What version of JupyterHub (git commit, if running from a repo)?

I think this is a regression in the connection-checking in 0.4. I believe I have a fix.

**Comment by minrk** [comment]
Wrong button...

**Comment by minrk** [comment]
Should be fixed by #453
